packages/nuxt/src/core/nuxt.ts

```diff
@@ -127,6 +127,12 @@ async function initNuxt (nuxt: Nuxt) {
     ...nuxt.options._modules
   ]
 
+  // Automatically register user modules
+  for (const config of nuxt.options._layers.map(layer => layer.config)) {
+    const userModules = await resolveFiles(config.srcDir, `${config.dir?.modules || 'modules'}/*{${nuxt.options.extensions.join(',')}}`)
+    modulesToInstall.push(...userModules)
```

Comment:
- We need to run these modules **before** ad-hoc modules (_modules). Probably after user modules. 
- Can you check if there are sorted a-z? A explicit sort by name would be nice. 
- We might need to duplicate them too if explicitly added by user for custom order

Reply:
yes, they are sorted.
and they will be de-duplicated on run